src/client/camera.cpp

```diff
@@ -403,34 +404,37 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime, f32 tool_r
 		if (m_camera_mode == CAMERA_MODE_THIRD_FRONT)
 			m_camera_direction *= -1;
 
-		my_cp.Y += 2;
-
-		// Calculate new position
-		bool abort = false;
-		for (int i = BS; i <= BS * 2.75; i++) {
-			my_cp.X = m_camera_position.X + m_camera_direction.X * -i;
-			my_cp.Z = m_camera_position.Z + m_camera_direction.Z * -i;
-			if (i > 12)
-				my_cp.Y = m_camera_position.Y + (m_camera_direction.Y * -i);
-
-			// Prevent camera positioned inside nodes
-			const NodeDefManager *nodemgr = m_client->ndef();
-			MapNode n = m_client->getEnv().getClientMap()
-				.getNode(floatToInt(my_cp, BS));
-
-			const ContentFeatures& features = nodemgr->get(n);
-			if (features.walkable) {
-				my_cp.X += m_camera_direction.X*-1*-BS/2;
-				my_cp.Z += m_camera_direction.Z*-1*-BS/2;
-				my_cp.Y += m_camera_direction.Y*-1*-BS/2;
-				abort = true;
+		v3f shootline_end = my_cp - m_camera_direction * 2.75f * BS;
+		v3f shootline_extended = shootline_end - m_camera_direction * BS;
+
+		RaycastState ray(core::line3d<f32>(m_camera_position, shootline_extended),
+			true, false);
```

Comment:
So, how about `false, false` to remove object checking (which current code does not do)? I guess this depends on profiling.

Reply:
In a situation where objects are moving around near the player, perhaps it is better those objects do not move the camera about?